src/jit/codegenlinear.cpp

```diff
@@ -18,6 +18,104 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "emit.h"
 #include "codegen.h"
 
+//------------------------------------------------------------------------
+// genInitializeRegisterState: Clean the spills' descriptors of "regSet" and mark all the register in "regSet" that hold
+// an argument of the function (which is the first block).
+//
+// Assumptions:
+//    On exit the "rsModifiedRegsMask" (in "regSet") holds all the registers' masks hosting an argument on the function
+//    and elements of "rsSpillDesc" (in "regSet") are setted to nullptr.
+//
+// Notes:
+//    This method is intended to be called only from initializeStructuresBeforeBlockCodeGeneration.
+void CodeGen::genInitializeRegisterState()
+{
+    // Initialize the spill tracking logic
+
+    regSet.rsSpillBeg();
+
+    // If any arguments live in registers, mark those regs as such
+
+    unsigned   varNum;
+    LclVarDsc* varDsc;
+
+    for (varNum = 0, varDsc = compiler->lvaTable; varNum < compiler->lvaCount; varNum++, varDsc++)
+    {
+        // Is this variable a parameter assigned to a register?
+        if (!varDsc->lvIsParam || !varDsc->lvRegister)
+        {
+            continue;
+        }
+
+        // Is the argument live on entry to the method?
+        if (!VarSetOps::IsMember(compiler, compiler->fgFirstBB->bbLiveIn, varDsc->lvVarIndex))
+        {
+            continue;
+        }
+
+        // Is this a floating-point argument?
+        if (varDsc->IsFloatRegType())
+        {
+            continue;
+        }
+
+        noway_assert(!varTypeIsFloating(varDsc->TypeGet()));
+
+        // Mark the register as holding the variable
+        assert(varDsc->lvRegNum != REG_STK);
+        if (!varDsc->lvAddrExposed)
+        {
+            regSet.verifyRegUsed(varDsc->lvRegNum);
+        }
+    }
+}
+
+//------------------------------------------------------------------------
+// genInitialize: Initialize Scopes, registers, gcInfo and current life variables structures
```

Comment:
To be more concise (and consistent with general use) I would say "current liveness structures" instead of "current life variables structures"

Reply:
Done